### PR TITLE
Fix wrong longtouch

### PR DESCRIPTION
**Pull request by yunikkk**
### Summary of changes

Testing fix of MAPSAND-720, MAPSAND-699 and #1897 

### User impact (optional)

Notice how new pin is added to the map before fix after closing the view annotation - meaning onLongPress was fired by the SDK without actual gesture.

Important notice - gestures that involve `MotionEvent.ACTION_UP` (double tap to zoom, tap and scroll to zoom) are still _not_ intercepted from ViewAnnotations. This works differently than on iOS, however implementation seems complicated, since to implement them we'll need to notify gesture handlers in `onInterceptTouchEvent` instead of `onTouchEvent`. 

| Before  | After |
| ------------- | ------------- |
| <video src="https://user-images.githubusercontent.com/2265924/214882613-9bcf659e-4698-4fc6-8fa7-96538f033165.mp4">  |  <video src="https://user-images.githubusercontent.com/2265924/214882709-ba3977fd-8a31-450f-92a9-29836cfbc96f.mp4"> |

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Comment by pengdev** [comment]
Can we add a few tests to cover the changes?

**Comment by yunikkk** [comment]
> Can we add a few tests to cover the changes?

Yes sure, still solving some issues with multitouch (e.g. 3 fingers gestures still fire longPress for some reason, may be an issue with the gestures lib and not ours)

File: app/src/androidTest/java/com/mapbox/maps/testapp/gestures/GestureInterceptedOnViewAnnotationTest.kt [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            cameraOptions {
+              center(cameraCenter)
+            }
+          )
+
+          mapView.viewAnnotationManager.apply {
+
+            // xml layout resource is only available within instrumentationContext,
+            // while mapView uses targetContext hence fails to inflate if used via resId
+            val viewAnnotation = LayoutInflater.from(InstrumentationRegistry.getInstrumentation().context).inflate(R.layout.view_annotation, mapView, false)
```
Comment (yunikkk) [inline app/src/androidTest/java/com/mapbox/maps/testapp/gestures/GestureInterceptedOnViewAnnotationTest.kt:69]: Weirdly `targetContext` fails to inflate the view_annotation that is added to the androidText source set, though it's working in the sdk androidTest e.g. https://github.com/mapbox/mapbox-maps-android/blob/main/sdk/src/androidTest/java/com/mapbox/maps/ViewAnnotationTest.kt. Guess activity is setup differently (part of instrumentation app and not the main one https://github.com/mapbox/mapbox-maps-android/blob/main/sdk/src/androidTest/java/com/mapbox/maps/EmptyActivity.kt]), decided to inflate explicitly with instrContext.

File: app/src/androidTest/java/com/mapbox/maps/testapp/gestures/GestureInterceptedOnViewAnnotationTest.kt [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+              }
+            )
+          }
+        }
+      }
+    }
+
+    // click on view annotation to close it
+    onView(withId(R.id.view_annotation)).perform(
+      GesturesUiTestUtils.move(
```
Comment (alexander-kulikovskii) [inline app/src/androidTest/java/com/mapbox/maps/testapp/gestures/GestureInterceptedOnViewAnnotationTest.kt:105]: just a question
click made with move action?

File: sdk/src/main/java/com/mapbox/maps/MapView.kt [hunk 3]
```diff
@@ -325,17 +336,28 @@ open class MapView : FrameLayout, MapPluginProviderDelegate, MapControllable {
...
+      MotionEvent.ACTION_UP,
+      MotionEvent.ACTION_CANCEL -> {
+        interceptedViewAnnotationEvents.clear()
+        return false
+      }
+      MotionEvent.ACTION_POINTER_UP -> {
+        val upPointerId = event.getPointerId(event.actionIndex)
+        interceptedViewAnnotationEvents.removeAll {
+          val cachedPointerId = it.getPointerId(it.actionIndex)
+          upPointerId == cachedPointerId
```
Comment (alexander-kulikovskii) [inline sdk/src/main/java/com/mapbox/maps/MapView.kt:356]:
```suggestion
          upPointerId == it.getPointerId(it.actionIndex)
```

File: app/src/androidTest/java/com/mapbox/maps/testapp/gestures/GestureInterceptedOnViewAnnotationTest.kt (hunk 2 above)
Comment (yunikkk) [reply app/src/androidTest/java/com/mapbox/maps/testapp/gestures/GestureInterceptedOnViewAnnotationTest.kt:105]: Yup, move delta is small so it's effectively a click :)

Comment (alexander-kulikovskii) [reply app/src/androidTest/java/com/mapbox/maps/testapp/gestures/GestureInterceptedOnViewAnnotationTest.kt:105]: also you can add some extension: 

```
R.id.view_annotation.click()
```

```
com.mapbox.maps.R.id.mapView.move(deltaX, deltaY)
```
and hide `onVIew(withId(...)).perform(...`

Comment (yunikkk) [reply app/src/androidTest/java/com/mapbox/maps/testapp/gestures/GestureInterceptedOnViewAnnotationTest.kt:105]: Added `click()` shorthand for move with arguments, however I wouldn't change the existing calls to extensions of Int, could be confusing and increases number of change if applied accross the codebase

**Review by ank27 — APPROVED** [review]

**Review by alexander-kulikovskii — APPROVED** [review]

**Pull request merged**